gulp-core-build/src/tasks/BuildReceiptTask.ts

```diff
@@ -0,0 +1,134 @@
+import { GulpTask } from './GulpTask';
+import gulp = require('gulp');
+import * as fs from 'fs';
+import * as path from 'path';
+import * as child_process from 'child_process';
+
+export interface IBuildReceiptTask {
+}
+
+const PROCESS_OUTPUT_DELIMITER: string = '~X~X~X~X~X~X~';
+
+let _lastLocalHashes: { [path: string]: string } = {};
+
+/**
+ * This task is responsible for the following:
+ *
+ * 1. Gather the local files build receipt of the current files. Cache it in _lastFilesHash for later use.
```

Comment:
"Gather the local files build receipt of...", "local files build receipt" is a little hard to parse...